README.md

```diff
@@ -76,3 +84,35 @@ end
 ```
 
 Obviously every table type is different, but via a combination of `Tables.rows` and `Tables.columns` each table type should be able to construct an instance of itself.
+
+## Functions that input and output tables:
+
+For functions that input a table, perform some calculation, and output a new table, we need a way of constructing the preferred output table given the input.  For this purpose, the `Tables.materializer(table)` function returns the the constructor for the preferred in-memory structure of the input table.  By default, `Tables.materializer(table)` returns `identity`.
```

Comment:
What guarantees do we want to provide? Should the table be mutable? It would be nice to be explicit about this.

If we want the table to be mutable, I still think `similar` would be a good function for this.

Reply:
I think `materializer` is different enough from `similar` that it would cause confusion.  `materializer(tbl)` does not return a table; it returns a function.

E.g.

```
materializer(x::ColumnTable) = columntable

materializer(x::DataFrame) = DataFrame
```

So that `materializer(tbl)(other_tbl)` creates the preferred table type of `tbl`.

Reply:
Ah, okay.  The assumption of a mutable table may be too strong.  Since sink functions are already encouraged by the interface, I'm partial to this current approach of "give me the sink function for this table".